src/node/plugin.ts

```diff
@@ -23,38 +27,58 @@ require("module")._load = function (request: string, parent: object, isMain: boo
   return originalLoad.apply(this, [request.replace(/^code-server/, path.resolve(__dirname, "../..")), parent, isMain])
 }
 
+/**
+ * Load a plugin and run its activation function.
+ */
 const loadPlugin = async (pluginPath: string, httpServer: HttpServer, args: Args): Promise<void> => {
   try {
     const plugin: Plugin = require(pluginPath)
     plugin.activate(httpServer, args)
-    logger.debug("Loaded plugin", field("name", path.basename(pluginPath)))
+
+    logger.debug(
+      "Loaded plugin",
+      field("name", path.basename(pluginPath)),
+      field("version", require(path.join(pluginPath, "package.json")).version || "n/a"),
+    )
   } catch (error) {
-    if (error.code !== "MODULE_NOT_FOUND") {
-      logger.warn(error.message)
-    } else {
-      logger.error(error.message)
-    }
+    logger.error(error.message)
   }
 }
 
-const _loadPlugins = async (httpServer: HttpServer, args: Args): Promise<void> => {
-  const pluginPath = path.resolve(__dirname, "../../plugins")
-  const files = await util.promisify(fs.readdir)(pluginPath, {
-    withFileTypes: true,
-  })
-  await Promise.all(files.map((file) => loadPlugin(path.join(pluginPath, file.name), httpServer, args)))
-}
-
-export const loadPlugins = async (httpServer: HttpServer, args: Args): Promise<void> => {
+/**
+ * Load all plugins in the specified directory.
+ */
+const _loadPlugins = async (pluginDir: string, httpServer: HttpServer, args: Args): Promise<void> => {
   try {
-    await _loadPlugins(httpServer, args)
+    const files = await util.promisify(fs.readdir)(pluginDir, {
+      withFileTypes: true,
+    })
+    await Promise.all(files.map((file) => loadPlugin(path.join(pluginDir, file.name), httpServer, args)))
   } catch (error) {
     if (error.code !== "ENOENT") {
       logger.warn(error.message)
     }
   }
+}
 
-  if (process.env.PLUGIN_DIR) {
-    await loadPlugin(process.env.PLUGIN_DIR, httpServer, args)
-  }
+/**
+ * Load all plugins from the `plugins` directory, directories specified by
+ * `CS_PLUGIN_PATH` (colon-separated), and individual plugins specified by
+ * `CS_PLUGIN` (also colon-separated).
+ */
+export const loadPlugins = async (httpServer: HttpServer, args: Args): Promise<void> => {
+  const pluginPath = process.env.CS_PLUGIN_PATH || `${path.join(paths.data, "plugins")}:/etc/code-server/plugins`
```

Comment:
Since we're using the data dir, we probably want to use `/usr/share/code-server/plugins` instead of `/etc/code-server/plugins`.

See https://wiki.archlinux.org/index.php/XDG_Base_Directory

Reply:
Ahhh good point. I see now your original comment used the config dir, do you think that would be more appropriate than the data dir? My instinct was to use data but I'm not sure if that's how most editors do it.

Reply:
data-dir makes most sense as plugins won't be user editable.